src/test/munit/directory-listener-functional-test-case.xml

```diff
@@ -0,0 +1,396 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<mule xmlns:validation="http://www.mulesoft.org/schema/mule/validation"
+      xmlns:file="http://www.mulesoft.org/schema/mule/file"
+      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+      xmlns:munit="http://www.mulesoft.org/schema/mule/munit"
+      xmlns:munit-tools="http://www.mulesoft.org/schema/mule/munit-tools"
+      xmlns="http://www.mulesoft.org/schema/mule/core"
+      xsi:schemaLocation="
+		http://www.mulesoft.org/schema/mule/validation http://www.mulesoft.org/schema/mule/validation/current/mule-validation.xsd
+		http://www.mulesoft.org/schema/mule/core http://www.mulesoft.org/schema/mule/core/current/mule.xsd
+		http://www.mulesoft.org/schema/mule/munit http://www.mulesoft.org/schema/mule/munit/current/mule-munit.xsd
+		http://www.mulesoft.org/schema/mule/munit-tools  http://www.mulesoft.org/schema/mule/munit-tools/current/mule-munit-tools.xsd
+		http://www.mulesoft.org/schema/mule/file http://www.mulesoft.org/schema/mule/file/current/mule-file.xsd">
+
+    <munit:config name="directory-listener-functional-test-case"/>
+
+    <file:matcher name="Matcher" filenamePattern='matchme.txt'/>
+
+    <munit:before-test name="directory-listener-functional-test-case-Before_Test" description="Before tests actions">
+        <munit-tools:store  key="messages">
+            <munit-tools:value >#[[]]</munit-tools:value>
+        </munit-tools:store>
+        <file:create-directory config-ref="file-list-config" directoryPath="matcherless"/>
+        <file:create-directory config-ref="file-list-config" directoryPath="withMatcher"/>
+    </munit:before-test>
+
+    <sub-flow name="storeMessages">
+        <set-variable value="#[message]" variableName="message"/>
+        <munit-tools:retrieve  key="messages"/>
+        <munit-tools:store key="messages">
+            <munit-tools:value >#[
+                %dw 2.0
+                output application/java
+                ---
+                payload ++ [vars.message]
+                ]
+            </munit-tools:value>
+        </munit-tools:store>
+    </sub-flow>
+
+    <flow name="source-on-file-created-test" >
+        <file:listener config-ref="file-list-config" directory="matcherless" recursive="false">
+            <scheduling-strategy >
+                <fixed-frequency />
+            </scheduling-strategy>
+        </file:listener>
+        <flow-ref name="storeMessages"/>
+    </flow>
+    <munit:test name="on-file-created-test" description="Verifies that a created file is picked">
+        <munit:enable-flow-sources >
+            <munit:enable-flow-source value="source-on-file-created-test"/>
+        </munit:enable-flow-sources>
+        <munit:behavior >
+            <file:write config-ref="file-list-config" path="matcherless/watchme.txt">
+                <file:content>who watches the watchmen?</file:content>
```

Comment:
It would be great if there was a way to put the names of the files and the content in variables so that is not hardcoded.